src/meta_memcache/base/cache_pool.py

```diff
@@ -0,0 +1,251 @@
+import time
+from typing import Any, Optional, Tuple, Type, TypeVar
+
+from meta_memcache.base.base_cache_pool import BaseCachePool
+from meta_memcache.configuration import LeasePolicy, RecachePolicy, StalePolicy
+from meta_memcache.errors import MemcacheError
+from meta_memcache.protocol import Flag, IntFlag, Key, Miss, Success, Value
+
+T = TypeVar("T")
+
+
+class CachePool(BaseCachePool):
+    def set(
+        self,
+        key: Key,
+        value: Any,  # pyre-ignore[2]
+        ttl: int,
+        no_reply: bool = False,
+        cas: Optional[int] = None,
+        stale_policy: Optional[StalePolicy] = None,
+    ) -> bool:
+        flags = set()
+        if no_reply:
+            flags.add(Flag.NOREPLY)
+        int_flags = {
+            IntFlag.CACHE_TTL: ttl,
+        }
+        if cas is not None:
+            int_flags[IntFlag.CAS] = cas
+            if stale_policy and stale_policy.mark_stale_on_cas_mismatch:
+                flags.add(Flag.MARK_STALE)
+
+        result = self.ms(
+            key=key,
+            value=value,
+            ttl=ttl,
+            flags=flags,
+            int_flags=int_flags,
+        )
+
+        return isinstance(result, Success)
+
+    def delete(
+        self,
+        key: Key,
+        cas: Optional[int] = None,
+        no_reply: bool = False,
+        stale_policy: Optional[StalePolicy] = None,
+    ) -> bool:
+        flags = set()
+        int_flags = {}
+        if no_reply:
+            flags.add(Flag.NOREPLY)
+        if cas is not None:
+            int_flags[IntFlag.CAS] = cas
+        if stale_policy and stale_policy.mark_stale_on_deletion_ttl > 0:
+            flags.add(Flag.MARK_STALE)
+            int_flags[IntFlag.CACHE_TTL] = stale_policy.mark_stale_on_deletion_ttl
+
+        result = self.md(
+            key=key,
+            flags=flags,
+            int_flags=int_flags,
+        )
+
+        return isinstance(result, Success)
+
+    def touch(self, key: Key, ttl: int, no_reply: bool = False) -> bool:
+        flags = set()
+        int_flags = {IntFlag.CACHE_TTL: ttl}
+        if no_reply:
+            flags.add(Flag.NOREPLY)
+        result = self.mg(key, flags=flags, int_flags=int_flags)
```

Comment:
Touch is a meta get? I thought it was a multi set

Reply:
meta commands don't have multiget (you just send several mg without waiting).

And `touch` is implemented with meta get, yes, `mg` can change TTL, and `mg` can not return value, and can also have not reply at all.